api/pkg/apis/v1alpha1/providers/states/k8s/k8s.go

```diff
@@ -220,10 +231,49 @@ func (s *K8sStateProvider) Upsert(ctx context.Context, entry states.UpsertReques
 			sLog.Errorf("  P (K8s State): failed to get object: %v", err)
 			return "", err
 		}
+
+		if refreshLabels {
```

Comment:
I'm against maintaining the "latest" tag like this. I think we are mixing responsibilities of objects. The [Object] objects should behave exactly as before. Version resolutions (like [Object]:version) and understanding of "latest" should be in the new [Object]Container object.